inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/table/DynamicKafkaDeserializationSchema.java

```diff
@@ -128,24 +128,25 @@ public RowData deserialize(ConsumerRecord<byte[], byte[]> record) throws Excepti
     @Override
     public void deserialize(ConsumerRecord<byte[], byte[]> record, Collector<RowData> collector)
             throws Exception {
+        if (random.nextInt(20000000) < 10) {
+            throw new Exception("this is a test exception,--------");
+        }
         // shortcut in case no output projection is required,
         // also not for a cartesian product with the keys
         if (keyDeserialization == null && !hasMetadata) {
             valueDeserialization.deserialize(record.value(),
-                    sourceMetricData == null ? collector : new MetricsCollector<>(collector, sourceMetricData));
+                    sourceExactlyMetric == null ? collector : new MetricsCollector<>(collector, sourceExactlyMetric));
             return;
         }
-
         // buffer key(s)
         if (keyDeserialization != null) {
             keyDeserialization.deserialize(record.key(), keyCollector);
         }
         // project output while emitting values
         outputCollector.inputRecord = record;
         outputCollector.physicalKeyRows = keyCollector.buffer;
-        if (sourceMetricData != null) {
-            MetricsCollector<RowData> metricsCollector = new MetricsCollector<>(collector, sourceMetricData);
-            metricsCollector.resetTimestamp(getRecordTime(outputCollector.metadataConverters, record));
+        if (sourceExactlyMetric != null) {
+            MetricsCollector<RowData> metricsCollector = new MetricsCollector<>(collector, sourceExactlyMetric);
             outputCollector.outputCollector = metricsCollector;
```

Comment:
Local variable 'metricsCollector' is redundant

Reply:
fix it, thx~